Turn Architecture section intro into an overview

ptgott opened the pull request:
Closes #46015

While the Teleport docs do not include an architecture overview page, there is a section intro for `docs/pages/reference/architecture` that includes a table of contents. This change turns that page into an architectural overview, providing context around each link to a child architecture page.

ptgott committed:
Turn Architecture section intro into an overview

Closes #46015

While the Teleport docs do not include an architecture overview page,
there is a section intro for `docs/pages/reference/architecture` that
includes a table of contents. This change turns that page into an
architectural overview, providing context around each link to a child
architecture page.

zmb3 commented on docs/pages/reference/architecture/architecture.mdx:
```diff
@@ -1,10 +1,131 @@
+
+Teleport consists of a **control plane** that runs the Teleport Auth Service and
+Teleport Proxy Service, and a **data plane** that consists of Teleport Agents,
+Machine ID Bots, and Auth Service API clients like the Event Handler and Access
```
```suggestion
Machine ID Bots, and API clients like the Event Handler and Access
```

zmb3 commented on docs/pages/reference/architecture/architecture.mdx:
```diff
@@ -1,10 +1,131 @@
+Teleport Proxy Service, and a **data plane** that consists of Teleport Agents,
+Machine ID Bots, and Auth Service API clients like the Event Handler and Access
+Request plugins. The data plane communicates with both your infrastructure and
+the control plane. Your infrastructure communicates only with the data plane.
```
I'm not sure I find the distinction of the "data plane" helpful here.

Teleport agents run on the customer infrastructure, but they absolutely communicate with the control plane.

I think we can simplify this whole section by just defining the term _control plane_ as the auth+proxy services.

strideynet commented on docs/pages/reference/architecture/architecture.mdx:
```diff
@@ -1,10 +1,131 @@
+continuously refresh credentials. As with Agents, administrators must deploy
+`tbot` on their own infrastructure, including on CI/CD platforms such as GitHub
+Actions. The [join methods](../enroll-resources/machine-id/deployment.mdx) for
+`tbot` differ from those for Agents.
```
> The [join methods](../enroll-resources/machine-id/deployment.mdx) for
`tbot` differ from those for Agents.

The only join method not available to bots is the `ec2` join method since it was largely replaced by the `iam` join method.  All join methods availale for bots should also work for agents (we just don't tend to write guides for them as they don't make sense - e.g github actions).

ptgott committed:
Respond to feedback

- Tweak API wording.
- Remove the usage of "data plane".
- Correct item re: `tbot` join methods.
